### PR TITLE
PyTorch TEQ Weight-only 3x API Implementation

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: test/3x/torch/quantization/weight_only/test_teq.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                        "absorb_to_layer": {"transformer.h.0.mlp.fc_in": ["transformer.h.0.mlp.fc_out"]},
+                    },
+                },
+            }
+        }
+
+        qdq_model = quantize(model=self.gptj, quant_config=quant_config, run_fn=train, example_inputs=example_inputs)
+        self.assertTrue(isinstance(qdq_model, torch.nn.Module))
+        out2 = qdq_model(test_input)
+        self.assertTrue(torch.allclose(out1[0], out2[0]))
```
Comment (xin3he) [inline test/3x/torch/quantization/weight_only/test_teq.py:138]: @Kaihui-intel,  Please add one more check that compares fp32_out and woq_out.

**Review by xin3he — APPROVED** [review]

**Comment by xin3he** [comment]
LGTM

File: neural_compressor/torch/algorithms/weight_only/teq.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
+#
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/teq.py:4]:
```suggestion
# Copyright (c) 2024 Intel Corporation
```

Please also correct it in other recently added files as well.

File: neural_compressor/torch/algorithms/weight_only/teq.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+            torch.save(self.trained_alphas, save_scale_file)
+
+        if save_state_dict_file:  # pragma: no cover
+            torch.save(self.model.state_dict(), save_state_dict_file)
+
+
+def teq_quantize(
+    model, weight_config={}, absorb_to_layer={}, folding=True, dataloader=None, calib_func=None, example_inputs=None
+):
+    """Run weight-only quantization with."""
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/teq.py:307]:
```suggestion
    """Run TEQ weight-only quantization."""
```

File: neural_compressor/torch/algorithms/weight_only/teq.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+            assert False, "Please provide dataloader to train."
+        teq_quantizer.train(dataloader)
+
+    # 3. apply scale to model
+    teq_quantizer.transform()
+
+    # 4. get quantized model
+    teq_quantizer.quantize()
+
+    # quantization_data = gptq_quantizer.execute_quantization()
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/teq.py:342]:
```suggestion
```

**Review by yiliu30 — APPROVED** [review]
Left some nit comments, please also check the docstring style to align with [google coding style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods). Others are LGTM.

**Pull request merged**